google/cloud/internal/curl_handle_factory.h

```diff
@@ -154,6 +155,8 @@ class PooledCurlHandleFactory : public CurlHandleFactory {
   std::deque<CurlMulti> multi_handles_;
   mutable std::mutex last_client_ip_address_mu_;
   std::string last_client_ip_address_;
+  std::atomic<std::size_t> active_handles_;
+  std::atomic<std::size_t> active_multi_handles_;
```

Comment:
Opt: It looks like these are manipulated within, or very close to regions already protected by their matching mutexes, so you may want to take advantage of that to remove the need for `std::atomic`.

Reply:
Done. And fixed a bug in the process.